### PR TITLE
fix: set hasPrecision to false when type is integer, closes #743

### LINKED ISSUE #743: Maximum is not inclusive
Hey guys, I've found a bug in [this line](https://github.com/json-schema-faker/json-schema-faker/blob/develop/src/lib/types/number.mjs#LL50C58-L50C58) of code. The problem is the fifth parameter (hasPrecision) which 
makes it impossible to generante a random integer from 0 to 1 from the following schema:

```
{
  "type": "integer",
  "minimum": 0,
  "maximum": 1
}
```

From the docs, if I don't provide a boolean true for `exclusiveMaximum`, it should be able to generate a integer 0 or 1. With that in mind, the line I provided will always make the flag hasPrecision to be true, and will end up entering [this conditional](https://github.com/json-schema-faker/json-schema-faker/blob/develop/src/lib/core/random.mjs#L91), which will call [getRandom](https://github.com/json-schema-faker/json-schema-faker/blob/develop/src/lib/core/random.mjs#L66) instead of [getRandomInteger](https://github.com/json-schema-faker/json-schema-faker/blob/develop/src/lib/core/random.mjs#L6), causing it to always return a 0.

To make matters worse, the schema bellow will always return 0 too! This happens because the `exclusiveMaximum` substracts 1 from the maximum, making it have the same results from above.

```
{
  "type": "integer",
  "minimum": 0,
  "maximum": 2,
  "exclusiveMaximum": true
}
```

### PR DESCRIPTION
**Pull request by rfoel**
This is a quick solution for the issue I addressed in #743. Instead of making the flag hasPrecision always true for numbers, we can set it to false to fix the problem.

### CODE REVIEW

**Comment by pateketrueke** [comment]
Hi, so much thanks for your effort! Would you mind on adding a single test-case for this?

**Comment by rfoel** [comment]
Sure, will work on it.

File: tests/unit/generators/number.spec.mjs [hunk 1]
```diff
@@ -27,6 +27,11 @@ describe('Number Generator', () => {
     expect(isMultipleOf({ multipleOf: 8, minimum: 80, maximum: 90 })).to.be.true;
   });
 
+  it('should return an integer', () => {
+    const n = numberType({ type: 'integer' });
+    expect(Number.isInteger(n)).to.be.true;
```
Comment (pateketrueke) [inline tests/unit/generators/number.spec.mjs:32]: Good enough, thank you!

**Review by pateketrueke — APPROVED** [review]

**Comment by rfoel** [comment]
@pateketrueke I thought about testing it like this, it would really test what my changes fixed.
```js
  it('should return 1', () => {
    let n = numberType({ type: 'integer' });
    while (n !== 1) {
      n = numberType({ type: 'integer', minimum: 0, maximum: 1 });
    }
    expect(n).to.eq(1);
  });
```

**Comment by pateketrueke** [comment]
Oh, if that test works too I would add it as well, no problem!

**Comment by rfoel** [comment]
There you go https://github.com/json-schema-faker/json-schema-faker/pull/745

**Pull request merged**